pkg/storage/unified/resource/client.go

```diff
@@ -102,6 +150,45 @@ func idTokenExtractor(ctx context.Context) (string, error) {
 	return "", fmt.Errorf("id-token not found")
 }
 
+func namespaceExtractor(ctx context.Context) (string, []string, error) {
+	// Using identity.Requester instead of claims.AuthInfo because Namespace() relies on AllowedKubernetesNamespace, which is empty.
+	caller, err := identity.GetRequester(ctx)
+	if err != nil {
+		return "", nil, err
+	}
+
+	namespace := caller.GetAllowedKubernetesNamespace()
```

Comment:
I'm still puzzled by this function :thinking: 
How is it different from the caller's namespace?

Reply:
This code was removed from this PR. We'll deal with this question in a separate PR.